java-checks/src/main/java/org/sonar/java/checks/security/CookieHttpOnlyCheck.java

```diff
@@ -339,11 +339,8 @@ private void updateIssuesToReport(MethodInvocationTree mit) {
       // ignore XSRF-TOKEN cookies
       return;
     }
-    if (setterArgumentHasCompliantValue(mit.arguments())) {
-      variablesToReport.remove(reference);
-    } else if (compliantConstructorInitializations.contains(reference)) {
```

Comment:
By removing this check, we can completely get rid of `compliantConstructorInitializations`, that was used only here!